pkg/fleet-manager/backup_controller.go

```diff
@@ -0,0 +1,291 @@
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"sigs.k8s.io/cluster-api/util/patch"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	backupapi "kurator.dev/kurator/pkg/apis/backups/v1alpha1"
+)
+
+// BackupManager reconciles a Backup object
+type BackupManager struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (b *BackupManager) SetupWithManager(ctx context.Context, mgr ctrl.Manager, options controller.Options) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&backupapi.Backup{}).
+		WithOptions(options).
+		Complete(b)
+}
+
+func (b *BackupManager) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
+	log := ctrl.LoggerFrom(ctx)
+	backup := &backupapi.Backup{}
+
+	if err := b.Client.Get(ctx, req.NamespacedName, backup); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("backup object not found", "backup", req)
+			return ctrl.Result{}, nil
+		}
+
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	// Initialize patch helper
+	patchHelper, err := patch.NewHelper(backup, b.Client)
+	if err != nil {
+		return ctrl.Result{}, errors.Wrapf(err, "failed to init patch helper for backup %s", req.NamespacedName)
+	}
+	// Setup deferred function to handle patching the object at the end of the reconciler
+	defer func() {
+		patchOpts := []patch.Option{}
+		if err := patchHelper.Patch(ctx, backup, patchOpts...); err != nil {
+			reterr = utilerrors.NewAggregate([]error{reterr, errors.Wrapf(err, "failed to patch %s  %s", backup.Name, req.NamespacedName)})
+		}
+	}()
+
+	// Check and add finalizer if not present
+	if !controllerutil.ContainsFinalizer(backup, BackupFinalizer) {
+		controllerutil.AddFinalizer(backup, BackupFinalizer)
+		return ctrl.Result{}, nil
+	}
+
+	// Handle deletion
+	if backup.GetDeletionTimestamp() != nil {
+		return b.reconcileDeleteBackup(ctx, backup)
+	}
+
+	// Handle the main reconcile logic
+	return b.reconcileBackup(ctx, backup)
+}
+
+// reconcileBackup handles the main reconcile logic for a Backup object.
+func (b *BackupManager) reconcileBackup(ctx context.Context, backup *backupapi.Backup) (ctrl.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Fetch destination clusters
+	destinationClusters, err := fetchDestinationClusters(ctx, b.Client, backup.Namespace, backup.Spec.Destination)
+	if err != nil {
+		log.Error(err, "failed to fetch destination clusters for backup", "backupName", backup.Name)
+		return ctrl.Result{}, err
+	}
+	// Apply velero backup resource in target clusters
+	result, err := b.reconcileBackupResources(ctx, backup, destinationClusters)
+	if err != nil || result.Requeue || result.RequeueAfter > 0 {
```

Comment:
where do you set requeue and requeueAfter

Reply:
I will fix it